### PR TITLE
Adding OpenACC support for Makefiles (Makefile.kokkos and Makefile.targets)

### PR DESCRIPTION
**Pull request by pedrovalerolara**
Only two files were modified Makefile.kokkos and Makefile.targets.
I tried to use the same style used for the other back-ends.
I made some tests running simple kernels. Everything worked well.

### CODE REVIEW

File: Makefile.kokkos [hunk 1]
```diff
@@ -159,7 +163,9 @@ KOKKOS_CXX_VERSION                   := $(strip $(shell $(CXX) --version       2
 KOKKOS_INTERNAL_COMPILER_INTEL       := $(call kokkos_has_string,$(KOKKOS_CXX_VERSION),Intel Corporation)
 KOKKOS_INTERNAL_COMPILER_XL          := $(strip $(shell $(CXX) -qversion       2>&1 | grep -c XL))
 KOKKOS_INTERNAL_COMPILER_CRAY        := $(strip $(shell $(CXX) -craype-verbose 2>&1 | grep -c "CC-"))
+KOKKOS_INTERNAL_COMPILER_PGI         := $(call kokkos_has_string,$(KOKKOS_CXX_VERSION),PGI)
```
Comment (dalg24) [inline Makefile.kokkos:166]: Did you mean to commit this?

File: Makefile.kokkos [hunk 2]
```diff
@@ -279,6 +285,18 @@ else
   KOKKOS_INTERNAL_OPENMPTARGET_FLAG := -fopenmp -foffload=nvptx-none
 endif
 
+# Set OpenACC flags.
+ifeq ($(KOKKOS_INTERNAL_COMPILER_PGI), 1)
+  KOKKOS_INTERNAL_OPENACC_FLAG := -O3 -acc
+else
+    ifeq ($(KOKKOS_INTERNAL_COMPILER_NVHPC), 1)
+      KOKKOS_INTERNAL_OPENACC_FLAG := -O3 -acc
```
Comment (dalg24) [inline Makefile.kokkos:293]: Why `-O3`?

Comment (pedrovalerolara) [reply Makefile.kokkos]: Yes, this is not the place for -O3. My fault!

File: Makefile.kokkos [hunk 3]
```diff
@@ -279,6 +284,18 @@ else
   KOKKOS_INTERNAL_OPENMPTARGET_FLAG := -fopenmp -foffload=nvptx-none
 endif
 
+# Set OpenACC flags.
+ifeq ($(KOKKOS_INTERNAL_COMPILER_PGI), 1)
+  KOKKOS_INTERNAL_OPENACC_FLAG := -acc
+else
+    ifeq ($(KOKKOS_INTERNAL_COMPILER_NVHPC), 1)
+      KOKKOS_INTERNAL_OPENACC_FLAG := -acc
```
Comment (JBludau) [inline Makefile.kokkos:292]: if these two do the same thing why not have them in the same group?

File: Makefile.kokkos [hunk 4]
```diff
@@ -279,6 +284,18 @@ else
...
+    else
+      #Assume GCC
+      KOKKOS_INTERNAL_OPENACC_FLAG := -fopenacc -foffload=nvptx-none
```
Comment (JBludau) [inline Makefile.kokkos:295]: hmm ... what about checking and printing if not gcc?

File: Makefile.kokkos (hunk 3 above)
Comment (pedrovalerolara) [reply Makefile.kokkos]: I removed the KOKKOS_INTERNAL_COMPILER_PGI lines.
In fact NVHPC and PGI is the same, I removed that in one of the previous comments that I forgot to remove this too.

File: Makefile.kokkos (hunk 4 above)
Comment (pedrovalerolara) [reply Makefile.kokkos]: Yes, sure! It is a good point, but what would be a good way to do that?
Maybe:
else
    $(error Makefile.kokkos: OpenACC is enabled but the compiler is neither NVHPC nor GCC (got version string $(KOKKOS_CXX_VERSION)) )
endif

Comment (JBludau) [reply Makefile.kokkos]: sounds good.

File: Makefile.targets [hunk 5]
```diff
@@ -107,5 +107,14 @@ Kokkos_OpenMPTarget_Task.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/OpenMP
 	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/OpenMPTarget/Kokkos_OpenMPTarget_Task.cpp
 endif
 
+ifeq ($(KOKKOS_INTERNAL_USE_OPENACC), 1)
+Kokkos_OpenACC.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC.cpp
+	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC.cpp
+Kokkos_OpenACCSpace.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACCSpace.cpp
+	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACCSpace.cpp
+Kokkos_OpenACC_Instance.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC_Instance.cpp
+	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC_Instance.cpp
```
Comment (dalg24) [inline Makefile.targets:116]:
```suggestion
Kokkos_OpenACC_SharedAllocationRecord.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC_SharedAllocationRecord.cpp
	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/OpenACC/Kokkos_OpenACC_SharedAllocationRecord.cpp
```

**Comment by dalg24** [comment]
Must keep an eye on #5451 which introduces a new `KOKKOS_ENABLE_OPENACC_COLLAPSE_HIERARCHICAL_CONSTRUCTS` macro

**Review by masterleinad — APPROVED** [review]
This doesn't do anything with `core/unit_test/Makefile` but that is pretty outdated anyway and we should decide if we want to keep it in sync with `core/unit_test/CMakeLists.txt` (or delete it saying that testing can only be done with `CMake`).

File: Makefile.kokkos (hunk 4 above)
Comment (seyonglee) [reply Makefile.kokkos]: The current OpenACC backend does not support GCC; please remove the GCC option lines and replace with the error message.

Comment (pedrovalerolara) [reply Makefile.kokkos]: Yes, I know. I wanted to keep GCC, although the current support of GCC for OpenACC C++ is not very good right now, maybe in a future can improve. Anyway, I can remove that for the moment, no problem.

**Review by seyonglee — APPROVED** [review]
It looks good to me.

**Review by dalg24 — CHANGES_REQUESTED** [review]
Breaks the CI

**Comment by seyonglee** [comment]
> Breaks the CI

Fixed the CI error.

File: Makefile.kokkos [hunk 6]
```diff
@@ -490,6 +504,11 @@ ifeq ($(KOKKOS_INTERNAL_USE_OPENMPTARGET), 1)
   endif
 endif
 
+ifeq ($(KOKKOS_INTERNAL_USE_OPENACC), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_OPENACC")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_OPENACC_COLLAPSE_HIERARCHICAL_CONSTRUCTS")
```
Comment (masterleinad) [inline Makefile.kokkos:509]: This is defined in the source file now, right?
```suggestion
```

**Review by masterleinad — CHANGES_REQUESTED** [review]

File: Makefile.kokkos [hunk 7]
```diff
@@ -279,6 +284,15 @@ else
...
 endif
 
+ifeq ($(KOKKOS_INTERNAL_USE_OPENACC), 1)
+# Set OpenACC flags.
+ifeq ($(KOKKOS_INTERNAL_COMPILER_NVHPC), 1)
+  KOKKOS_INTERNAL_OPENACC_FLAG := -acc
+else
+  $(error Makefile.kokkos: OpenACC is enabled but the compiler must be NVHPC (got version string $(KOKKOS_CXX_VERSION)))
+endif
+endif
```
Comment (dalg24) [inline Makefile.kokkos:294]: Indentation is wrong but I can live with it

**Review by dalg24 — APPROVED** [review]

File: Makefile.kokkos (hunk 7 above)
Comment (pedrovalerolara) [reply Makefile.kokkos]: Done!

**Pull request merged**